### ISSUE TITLE
Address blacklight deprecations

### ISSUE THREAD

**Issue by sandbergja** [pulibrary/DSS#310]
There are several blacklight-related deprecations when you run the test suite (copied below).  Many of them come from pul_assets, so it might make sense to do #256 first.

- [x] `render_hash_as_hidden_fields`
- [x] `search_fields`
- [x] `search_field_options_for_select`
- [x] `should_autofocus_on_search_box?`
- [x] `facet_group_names`
- [x] Blacklight 8 will filter out non-search parameter
- [x] `query_has_constraints?`
- [x] `start_over_path`
- [x] `constraint_query_label`
- [x] `render_document_main_content_partial`

```
DEPRECATION WARNING: render_hash_as_hidden_fields is deprecated and will be removed from a future release (Use Blacklight::HiddenSearchStateComponent instead). (called from block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720 at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:2) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:2:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/form_tag_helper.rb:78:in `form_tag'

DEPRECATION WARNING: search_fields is deprecated and will be removed from a future release (removed without replacement). (called from block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720 at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:4) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:4:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/form_tag_helper.rb:78:in `form_tag'

DEPRECATION WARNING: search_field_options_for_select is deprecated and will be removed from a future release (removed without replacement). (called from search_fields at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/configuration_helper_behavior.rb:25) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/configuration_helper_behavior.rb:25:in `search_fields'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:4:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'

DEPRECATION WARNING: search_fields is deprecated and will be removed from a future release (removed without replacement). (called from block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720 at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:7) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:7:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/form_tag_helper.rb:78:in `form_tag'

DEPRECATION WARNING: search_field_options_for_select is deprecated and will be removed from a future release (removed without replacement). (called from search_fields at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/configuration_helper_behavior.rb:25) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/configuration_helper_behavior.rb:25:in `search_fields'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:7:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'

DEPRECATION WARNING: should_autofocus_on_search_box? is deprecated and will be removed from blacklight 8.0 (use SearchBarPresenter#autofocus?). (called from block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720 at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:14) (12 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/bundler/gems/pul_assets-e2b0508c2778/app/views/pul-assets/blacklight/catalog/_search_form.html.erb:14:in `block in ___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______bundler_gems_pul_assets_e_b____c_____app_views_pul_assets_blacklight_catalog__search_form_html_erb___3878774034004685416_147720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/capture_helper.rb:45:in `capture'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/helpers/form_tag_helper.rb:78:in `form_tag'

DEPRECATION WARNING: facet_group_names is deprecated and will be removed from a future release (Use blacklight_config.facet_group_names instead). (called from _app_views_catalog__facets_html_erb__3086858709878830743_157840 at /Users/sandbergj/repos/DSS/app/views/catalog/_facets.html.erb:1) (4 times); e.g.: 
    /Users/sandbergj/repos/DSS/app/views/catalog/_facets.html.erb:1:in `_app_views_catalog__facets_html_erb__3086858709878830743_157840'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `public_send'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `_run'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/template.rb:157:in `block in render'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4/lib/active_support/notifications.rb:208:in `instrument'

DEPRECATION WARNING: Blacklight 8 will filter out non-search parameter, including: utf8. (called from permit_search_params at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77) (1 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77:in `permit_search_params'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/search_state.rb:28:in `initialize'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/controller.rb:70:in `new'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/controller.rb:70:in `search_state'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/search_context.rb:53:in `find_search_session'

DEPRECATION WARNING: Blacklight 8 will filter out non-search parameter, including: utf8. (called from permit_search_params at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77) (1 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77:in `permit_search_params'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/search_state.rb:28:in `initialize'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/catalog_helper_behavior.rb:329:in `new'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/catalog_helper_behavior.rb:329:in `render_search_to_page_title'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/views/catalog/_search_results.html.erb:1:in `___sers_sandbergj__asdf_installs_ruby_______lib_ruby_gems_______gems_blacklight________app_views_catalog__search_results_html_erb__605319289569675499_167500'

DEPRECATION WARNING: query_has_constraints? is deprecated and will be removed from blacklight 8.0 (use search_state#has_constraints?). (called from _app_views_catalog__constraints_html_erb___2047367466655739718_167720 at /Users/sandbergj/repos/DSS/app/views/catalog/_constraints.html.erb:1) (1 times); e.g.: 
    /Users/sandbergj/repos/DSS/app/views/catalog/_constraints.html.erb:1:in `_app_views_catalog__constraints_html_erb___2047367466655739718_167720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `public_send'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `_run'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/template.rb:157:in `block in render'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4/lib/active_support/notifications.rb:208:in `instrument'

DEPRECATION WARNING: start_over_path is deprecated and will be removed from a future release (Removed without replacement). (called from _app_views_catalog__constraints_html_erb___2047367466655739718_167720 at /Users/sandbergj/repos/DSS/app/views/catalog/_constraints.html.erb:5) (1 times); e.g.: 
    /Users/sandbergj/repos/DSS/app/views/catalog/_constraints.html.erb:5:in `_app_views_catalog__constraints_html_erb___2047367466655739718_167720'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `public_send'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `_run'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/template.rb:157:in `block in render'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4/lib/active_support/notifications.rb:208:in `instrument'

DEPRECATION WARNING: constraint_query_label is deprecated and will be removed from a future release (Moving to Blacklight::ConstraintsComponent). (called from block in render_constraints_query at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/render_constraints_helper_behavior.rb:56) (1 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/render_constraints_helper_behavior.rb:56:in `block in render_constraints_query'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/render_constraints_helper_behavior.rb:55:in `render_constraints_query'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/render_constraints_helper_behavior.rb:38:in `block in render_constraints'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/helpers/blacklight/render_constraints_helper_behavior.rb:37:in `render_constraints'
    /Users/sandbergj/repos/DSS/app/views/catalog/_constraints.html.erb:7:in `_app_views_catalog__constraints_html_erb___2047367466655739718_167720'

DEPRECATION WARNING: Blacklight 8 will filter out non-search parameter, including: id. (called from permit_search_params at /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77) (2 times); e.g.: 
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/parameters.rb:77:in `permit_search_params'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/lib/blacklight/search_state.rb:28:in `initialize'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/controller.rb:70:in `new'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/controller.rb:70:in `search_state'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/blacklight-7.33.0/app/controllers/concerns/blacklight/searchable.rb:16:in `search_service'

DEPRECATION WARNING: render_document_main_content_partial is deprecated and will be removed from blacklight 8.0 (Use "render 'show_main_content'" instead). (called from _app_views_catalog_show_html_erb___1106922088946515340_221300 at /Users/sandbergj/repos/DSS/app/views/catalog/show.html.erb:5) (2 times); e.g.: 
    /Users/sandbergj/repos/DSS/app/views/catalog/show.html.erb:5:in `_app_views_catalog_show_html_erb___1106922088946515340_221300'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `public_send'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/base.rb:244:in `_run'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/actionview-7.0.4/lib/action_view/template.rb:157:in `block in render'
    /Users/sandbergj/.asdf/installs/ruby/3.1.3/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4/lib/active_support/notifications.rb:208:in `instrument'
```

**Comment by sandbergja** [comment]
FYI, here is the procedure I followed for the non-search parameters deprecation: https://github.com/pulibrary/orangelight/issues/3338#issue-1517815840

**Comment by maxkadel** [comment]
Closed by #322 and #324
